code/workspaces/infrastructure-api/src/kubernetes/configMapApi.spec.js

```diff
@@ -28,9 +28,9 @@ describe('namespacedConfigMapExists', () => {
     ).toEqual(true);
   });
 
-  it('returns false if reading config map is unsuccessful', async () => {
+  it('returns false if reading config map returns 404', async () => {
     k8sApiMock.readNamespacedConfigMap.mockImplementationOnce(
-      () => { throw new Error('expected test error'); },
+      () => { throw { response: { statusCode: 404 } }; }, // eslint-disable-line no-throw-literal
```

Comment:
Would it be cleaner to throw a K8sApiError, rather than a literal?

Reply:
This is mirroring how the k8s API throws errors which then get caught and wrapped in the K8sApiError class by the function that calls the API so I think this needs to be a literal. It would be good if I could make that more clear in the test however!